saos-enrichment/src/test/java/pl/edu/icm/saos/enrichment/apply/DefaultEnrichmentTagApplierTest.java

```diff
@@ -0,0 +1,160 @@
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+
+import pl.edu.icm.saos.common.json.JsonStringParser;
+import pl.edu.icm.saos.persistence.enrichment.model.EnrichmentTag;
+import pl.edu.icm.saos.persistence.model.Judgment;
+
+/**
+ * @author Łukasz Dumiszewski
+ */
+
+public class DefaultEnrichmentTagApplierTest {
+
+    private static final String HANDLED_TAG_TYPE = "handledTagType";
+    
+ 
+    @Mock private JsonStringParser<String> jsonStringParser;
+    
+    @Mock private EnrichmentTagValueConverter<String, Object> enrichmentTagValueConverter;
+    
+    @Mock private JudgmentUpdater<Object> judgmentUpdater;
+    
+    @Mock private Judgment judgment;
+    
+    @Mock private EnrichmentTag enrichmentTag;
+ 
+    
+    private DefaultEnrichmentTagApplier<String, Object> defaultEnrichmentTagApplier = new DefaultEnrichmentTagApplier<>(HANDLED_TAG_TYPE);
+    
+    
+    
+    @Before
+    public void before() {
+        
+        initMocks(this);
+        
+        defaultEnrichmentTagApplier.setJsonStringParser(jsonStringParser);
+        defaultEnrichmentTagApplier.setEnrichmentTagValueConverter(enrichmentTagValueConverter);
+        defaultEnrichmentTagApplier.setJudgmentUpdater(judgmentUpdater);
+        
+        when(enrichmentTag.getTagType()).thenReturn(HANDLED_TAG_TYPE);
+        
+    }
+    
+    
+    //------------------------ TESTS --------------------------
+    
+    
+    @Test
+    public void handlesEnrichmentTagType_DoesNotHandle() {
+        
+        // execute & assert
+        assertFalse(defaultEnrichmentTagApplier.handlesEnrichmentTagType("XXX"));
+        
+        // assert
+        verifyZeroInteractions(enrichmentTagValueConverter, jsonStringParser, judgmentUpdater);
+
+    }
+    
+    @Test
+    public void handlesEnrichmentTagType_Handles() {
+        
+        // execute & assert
+        assertTrue(defaultEnrichmentTagApplier.handlesEnrichmentTagType(HANDLED_TAG_TYPE));
+        
+        // assert
+        verifyZeroInteractions(enrichmentTagValueConverter, jsonStringParser, judgmentUpdater);
+
+    }
+    
+    
+    @Test(expected=IllegalArgumentException.class)
+    public void applyEnrichmentTag_NotHandledTagType() {
+        
+        // given
+        when(enrichmentTag.getTagType()).thenReturn("ABC");
+        
+        // execute
+        defaultEnrichmentTagApplier.applyEnrichmentTag(judgment, enrichmentTag);
+        
+        // assert
+        verifyZeroInteractions(enrichmentTagValueConverter, jsonStringParser, judgmentUpdater);
```

Comment:
I'll correct it so it can be reached. 
To be quite honest, checking the interactions with inner services is maybe a kind of exaggeration in situations like that, but thanks to that I am convinced that the exception is thrown in the expected place.